halibot/halauth.py

```diff
@@ -0,0 +1,77 @@
+import json
+import logging
+
+def hasPermission(perm, reply=False):
+	def real_dec(func):
+		def wrapper(self, msg, *args, **kwargs):
+			if self._hal.auth.hasPermission(msg.origin, msg.author, perm):
+				func(self, msg, *args, **kwargs)
+			elif reply:
+				self.reply(msg, body="Permission Denied")
+		return wrapper
+	return real_dec
+
+class HalAuth():
+
+	def __init__(self):
+		self.perms = []
+		self.enabled = False
+		self.log = logging.getLogger("Auth")
+
+	# Load permission file, and set to enabled
+	def load_perms(self, path):
+		self.path = path
+
+		try:
+			with open(self.path, "r") as f:
```

Comment:
Permissions will potentially get very large, keeps from cluttering the main one. Also, I'm not 100% sure that permissions will stay JSON, since the data doesn't really lend itself to a key-value form well. (More of a value-value-value).